### PR TITLE
[query] Fix int overflow in generic lines and add error for size limit

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/ir/GenericLines.scala [hunk 1]
```diff
@@ -160,7 +160,15 @@ object GenericLines {
             // move scanned input from buf to lineData
             val n = bufPos - begin
             if (linePos + n > lineData.length) {
-              val newLineData = new Array[Byte]((linePos + n) * 2)
+              val copySize = linePos.toLong + n
+              if (copySize > Int.MaxValue)
```
Comment (arunbhat) [inline hail/src/main/scala/is/hail/expr/ir/GenericLines.scala:164]: Should this be copySize >= Int.MaxValue. 

scala> val newLineData = new Array[Byte](Int.MaxValue)
java.lang.OutOfMemoryError: Requested array size exceeds VM limit

File: hail/src/main/scala/is/hail/expr/ir/GenericLines.scala [hunk 2]
```diff
@@ -160,7 +160,15 @@ object GenericLines {
...
+                fatal(s"GenericLines: line size reached: cannot read a line with more than 2^31-1 bytes")
+              val newSize = Math.min(copySize * 2, Int.MaxValue).toInt
```
Comment (arunbhat) [inline hail/src/main/scala/is/hail/expr/ir/GenericLines.scala:166]: Similarly should this be (Int.MaxValue-1)?

File: hail/src/main/scala/is/hail/expr/ir/GenericLines.scala (hunk 1 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/GenericLines.scala:164]: Hmm. This is an implementation limit, not a language limit:
https://www.baeldung.com/java-arrays-max-size

The link above suggests Int.MaxValue - 8

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/GenericLines.scala]: Pushed up a fix.

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/GenericLines.scala]: thanks for the comment!

**Review by chrisvittal — APPROVED** [review]

**Pull request merged**